tools/spec-gen-sdk/src/automation/entrypoint.ts

```diff
@@ -1,120 +1,68 @@
 import { RestEndpointMethodTypes } from '@octokit/rest';
 import { Octokit } from '@octokit/rest';
-import {
-  BlobServiceClient,
-  ContainerClient,
-} from '@azure/storage-blob';
+import * as fs from 'fs';
 import * as winston from 'winston';
-import { getAuthenticatedOctokit, RepoKey } from '../utils/githubUtils';
+import { getAuthenticatedOctokit, getRepoKey, RepoKey } from '../utils/githubUtils';
 import {
   FailureType,
   setFailureType,
   WorkflowContext,
-  workflowFilterSdkMain,
   workflowInit,
   workflowMain
 } from './workflow';
-import { TriggerType } from '../types/TriggerType';
 import {
-  getBlobName,
   loggerConsoleTransport,
   loggerDevOpsTransport,
-  loggerStorageAccountTransport,
+  loggerFileTransport,
   loggerTestTransport,
-  loggerWaitToFinish,
   sdkAutoLogLevels
 } from './logging';
-import { sdkAutoReportStatus } from './reportStatus';
-import { SDKAutomationState } from './sdkAutomationState';
-import { DefaultAzureCredential } from '@azure/identity';
-import * as pkginfo from 'pkginfo';
+import path from 'path';
+import { generateReport } from './reportStatus';
+import { SpecConfig, SdkRepoConfig, getSpecConfig, specConfigPath } from '../types/SpecConfig';
+import { getSwaggerToSdkConfig, SwaggerToSdkConfig } from '../types/SwaggerToSdkConfig';
 
 interface SdkAutoOptions {
   specRepo: RepoKey;
-  pullNumber: number;
   sdkName: string;
   branchPrefix: string;
-  filterSwaggerToSdk?: boolean;
+  localSpecRepoPath: string;
+  localSdkRepoPath: string;
+  tspConfigPath?: string;
+  readmePath?: string;  
+  pullNumber?: number;
+  specCommitSha?: string;
+  specRepoHttpsUrl?: string;
+  workingFolder: string;
 
   github: {
     token?: string;
-    id?: number;
-    privateKey?: string;
     commentAuthorName?: string;
   };
 
-  storage: {
-    name: string;
-    prefix: string;
-    downloadCommand: string;
-    isPublic: boolean;
-  };
-
-  buildID?: string;
   runEnv: 'local' | 'azureDevOps' | 'test';
 }
 
+type SpecPrInfo = {
+  head: {owner: string; repo: string};
+  base: {owner: string; repo: string};
+}
+
 export type SdkAutoContext = {
   config: SdkAutoOptions;
   octokit: Octokit;
```

Comment:
Do we need OctoKit any longer what are we trying to access from github?

Reply:
Same as the above reply.